dashboard/app/models/plc/enrollment_unit_assignment.rb

```diff
@@ -75,26 +75,26 @@ def focus_area_stage_ids
   def summarize_progress
     summary = []
 
-    categories_for_stage = plc_course_unit.script.lessons.map(&:flex_category).uniq
+    lesson_groups_for_lesson = plc_course_unit.script.lessons.map {|lesson| lesson.lesson_group ? lesson.lesson_group.key : lesson.flex_category}.uniq
```

Comment:
If possible, can you put this logic (use lesson group key if present, else flex category) in a method on Lesson so it doesn't have to repeated? It looks like we do this in at least two other places in this PR. This would make it slightly easier to remove the fallback once we're no longer using it too. Or does the logic differ in each place so that this wouldn't really help?